annotations/views/project_views.py

```diff
@@ -60,6 +60,19 @@ def add_text(self, request, pk=None):
         serializer = ProjectSerializer(project)
         return Response(serializer.data)
 
+    @action(detail=True, methods=['DELETE'], url_name='deletetext')
+    def delete_text(self, request, pk=None):
+        text_id = request.data['text_id']
+
+        submitted = Appellation.objects.filter(occursIn_id=text_id, submitted=True)
+        if submitted:
+            return Response(status=status.HTTP_412_PRECONDITION_FAILED)
+        
+        project = get_object_or_404(TextCollection, pk=pk)
```

Comment:
It wouldn't let me add a comment in the front end but since we could possible return a 404 lets handle that in the front end as well. Right now we are only handling the 412.